presto-parser/src/main/java/io/prestosql/sql/tree/RowDataType.java

```diff
@@ -0,0 +1,137 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class RowDataType
+        extends DataType
+{
+    private final List<Field> fields;
+
+    public RowDataType(NodeLocation location, List<Field> fields)
+    {
+        super(Optional.of(location));
+        this.fields = ImmutableList.copyOf(fields);
+    }
+
+    public List<Field> getFields()
+    {
+        return fields;
+    }
+
+    @Override
+    public List<? extends Node> getChildren()
+    {
+        return fields;
+    }
+
+    @Override
+    protected <R, C> R accept(AstVisitor<R, C> visitor, C context)
+    {
+        return visitor.visitRowDataType(this, context);
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        io.prestosql.sql.tree.RowDataType that = (io.prestosql.sql.tree.RowDataType) o;
+        return fields.equals(that.fields);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(fields);
+    }
+
+    public static class Field
+            extends Node
+    {
+        private final Optional<Identifier> name;
+        private final DataType type;
+
+        public Field(NodeLocation location, Optional<Identifier> name, DataType type)
+        {
+            super(Optional.of(location));
+
+            this.name = requireNonNull(name, "name is null");
+            this.type = requireNonNull(type, "type is null");
+        }
+
+        public Optional<Identifier> getName()
+        {
+            return name;
+        }
+
+        public DataType getType()
+        {
+            return type;
+        }
+
+        @Override
+        public List<? extends Node> getChildren()
+        {
+            ImmutableList.Builder<Node> children = ImmutableList.builder();
+            name.ifPresent(children::add);
+            children.add(type);
+
+            return children.build();
+        }
+
+        @Override
+        protected <R, C> R accept(AstVisitor<R, C> visitor, C context)
+        {
+            return visitor.visitRowField(this, context);
+        }
+
+        @Override
+        public String toString()
+        {
+            return name + " " + type;
```

Comment:
I believe this will render with `Optional`.  Is that what you want?

Reply:
It's for debugging, so it doesn't really matter. But I'll update it to omit the name if missing.